### ISSUE TITLE
Treatment of non-master / non-nxs HDF5 files

### ISSUE THREAD

**Issue by graeme-winter** [cctbx/dxtbx#244]
Semi-frequent issue with Eiger data: user has

```
ins_1_1_000001.h5
ins_1_1_000002.h5
ins_1_1_000003.h5
ins_1_1_000004.h5
ins_1_1_master.h5
ins_1_1_meta.h5
```

and passes `ins_1_1_000001.h5` to `dials.import` - this obviously cannot work because there is no metadata associated with this file _however_ it appears that we _do_ have a format class which recognises this namely `FormatHDF5RawData.py:FormatRawData` which seems hard coded for:

 - HDF5 containing just `data`
 - `data` being a *one* image data set - if this is not a 1 image data set then world will end because

```
    def get_raw_data(self):
        data = self._h5_handle["data"]
        return flex.int(data[:, :])
```

ohai loading 18 gigapixels via numpy (`p(ends_well) = ε`). 

Questions - 
 - do we need this?
 - can I hack it to impose assertions which are encoded in there?
 - how _should_ we trap user passing incorrect data & help them to get the right thing?

**Comment by graeme-winter** [comment]
With change above now picks up `FormatHDF5` which is not really designed to be used directly (i.e. it is not an active format class) => add a flag or something to incomplete format classes which can indicate that this file should not be used directly? Then `Registry.py:get_format_class_for_file` can ignore it...

**Comment by graeme-winter** [comment]
As an addendum to this: being able to read `foobar_000001.h5` as a data file with metadata coming from `${SOMEWHERE}` could be a useful feature to have, provided we have a mechanism to inject that data somehow.

**Comment by graeme-winter** [comment]
<dang> turns out that we somehow depended on this "feature" 🙄 

```
    def test_can_read_headerless_h5_and_no_detector_is_present(dials_data):
        data_h5 = dials_data("vmxi_thaumatin").join("image_15799_data_000001.h5").strpath
```

now fails (because it should never have worked?)

**Comment by graeme-winter** [comment]
=> `format-hdf5-raw` branch

**Comment by graeme-winter** [comment]
See also: `get-metadata-from-somewhere` branch on `dials`

**Comment by dagewa** [comment]
This is a good example where having context-specific Format classes (guess this is for XFEL?) in a separate repository and installable by `dxtbx.install_format` would make sense. Perhaps dxtbx should _only_ have base class formats (i.e. `FormatHDF5` in this tree), and anything else is kept elsewhere and can be installed as the "synchrotron pack" or the "XFEL pack" etc.

**Comment by graeme-winter** [comment]
@dagewa I would put up a _slight_ resistance to moving the synchrotron formats out of `dxtbx` since that is what `dxtbx` was written for. But it would be slight, and something I would file under "I would need persuading, but I am open to persuasion" 

I'm not hostile to the viewpoints expressed herein. 

Certainly the "special use case for one experiment in 2018" format classes _almost certainly should not_ be in the main repo.

**Comment by ndevenish** [comment]
> With change above now picks up `FormatHDF5` which is not really designed to be used directly (i.e. it is not an active format class) => add a flag or something to incomplete format classes which can indicate that this file should not be used directly? Then `Registry.py:get_format_class_for_file` can ignore it...

We do have something _slightly_ in place to do this - but only seems to be active in datablock template importing e.g. https://github.com/cctbx/dxtbx/blob/6e9157360ee5dd335c5766f4ed39593a18d818e4/datablock.py#L393-L394

However, the only place this seems to be set is in [`FormatNexusExternalDataFile`](https://github.com/cctbx/dxtbx/blob/6e9157360ee5dd335c5766f4ed39593a18d818e4/format/FormatNexusExternalDataFile.py#L48-L49).

**Comment by Anthchirp** [comment]
I suppose you could have that...
If you have a format class that understands a file but declines responsibility (ie. `ignore==True`) then you can end up in a situation where you have an understanding format class that you can't instantiate with no understanding children. This would then have to be resolved by passing the image up to the parent class of the understanding format class.
In that case you would end up with a `Format()` instance. I have no idea what consequences that would have.

**Comment by phyy-nx** [comment]
Archaeology tells me I made `FormatHDF5RawData.py` as a one-off for some ESRF data stills that were collected using an early membrane chip.  I used the format class with an old program `dxtbx.image2pickle` to create a file with beam center/distance metadata added to it.

I don't think I've worked with it for some time.  It's interesting that it's tested by this `vmxi_thaumatin` dataset though, specifically as a format class that doesn't produce a detector model.
